api/ruxos_posix_api/src/imp/execve/mod.rs

```diff
@@ -1,29 +1,32 @@
-use core::ffi::c_char;
-
 mod auxv;
 mod load_elf;
 mod stack;
 
 use alloc::vec;
+use core::ffi::c_char;
 
 use crate::{
+    config,
     imp::stat::{sys_getgid, sys_getuid},
-    sys_getegid, sys_geteuid,
+    sys_getegid, sys_geteuid, sys_random,
+    utils::char_ptr_to_str,
 };
 
 /// int execve(const char *pathname, char *const argv[], char *const envp[] );
 pub fn sys_execve(pathname: *const c_char, argv: usize, envp: usize) -> ! {
```

Comment:
It is better to use types that conform to the posix-defined interface for types. `pub fn sys_execve(pathname: *const c_char, char *const argv[], char *const envp[])` will be better.